### PR TITLE
Incremental model show reason for on_schema_change fail failures

### PR DESCRIPTION
**Pull request by ilanbenb**
resolves issue #5504

### Description

When the incremental model fails, I do not get the context I need to easily fix my discrepancy.
Adding more verbosity.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

**Comment by ilanbenb** [comment]
Hello,
This is my first contribution to this repo.
This first commit is a preliminary proposition and should be a conversation opener.
Thank you,
Ilan

**Comment by ilanbenb** [comment]
> for

Done :)

**Review by ChenyuLInx — APPROVED** [review]
This looks good to me! @ilanbenb  Thanks for adding this!

**Comment by ChenyuLInx** [comment]
I will leave this to the adapter team to take a second pass and merge(assuming all tests passes, I don't see any reason they won't)

File: core/dbt/include/global_project/macros/materializations/models/incremental/on_schema_change.sql
Comment (jtcohen6) [inline core/dbt/include/global_project/macros/materializations/models/incremental/on_schema_change.sql:126]: @dbeatty10 @dataders Does either of you have thoughts on the specific language here?

Comment (dbeatty10) [reply core/dbt/include/global_project/macros/materializations/models/incremental/on_schema_change.sql:126]: Something like "Additional troubleshooting context:" could be a drop-in replacement for "Extra info for easy anomaly detection:"

Side note: it's a bit unfortunate that "schema" is both the 2nd level in a three-level `database.schema.relation` hierarchy as well as the result of [DDL](https://en.wikipedia.org/wiki/Data_definition_language) operations on a relation.

Comment (ilanbenb) [reply core/dbt/include/global_project/macros/materializations/models/incremental/on_schema_change.sql]: @dbeatty10 I updated the log message per your request. It looks better

**Comment by ilanbenb** [comment]
Hey, @VersusFacit @jtcohen6 This pr is stall for more than 15 days.
Can it be reviewed by the code owners so it can be merged in the near future?

Thanks,
Ilan

**Comment by ChenyuLInx** [comment]
@ilanbenb  sorry for the delay! I am merging this for the the Adapter team as all feedback has been addressed. Thanks for the contribution!

**Pull request merged**
